src/llvm/di_builder.cr

```diff
@@ -36,11 +36,11 @@ struct LLVM::DIBuilder
   end
 
   def create_auto_variable(scope, name, file, line, type, align_in_bits)
-    LibLLVMExt.di_builder_create_auto_variable(self, scope, name, file, line, type, 0, DIFlags::Zero, align_in_bits)
+    LibLLVMExt.di_builder_create_auto_variable(self, scope, name, file, line, type, 1, DIFlags::Zero, align_in_bits)
```

Comment:
Could you explain why 1 instead of 0?

I ask because I have no idea what this `always_preserve` parameter means, but it would be really nice to have a small comment here explaining what `always_preserve` means and why we need to pass 1.

Reply:
Here is the comment from LLVM code about this parameter:
 // The optimizer may remove local variables. If there is an interest
    // to preserve variable info in such situation then stash it in a
    // named mdnode.

So I wanted to be sure that debug info will have all necessary info. We still losing some of the local variables at this moment. For example in compiler in normalizer.cr line 45 variable new_exp is not stored in DWARF file. I will investigate that further.

Reply:
Shouldn't it be `1` only when compiling in no release mode? Possibly this will prevent some optimization.

Reply:
Good point. I was planning to make it that way but it was skipping through the cracks.
I will do it that way: If it is full debug (with --debug flag) then it will be 0, otherwise it will be 1 as we still need to inject line numbers debug info unless --no-debug flag is provided)

Reply:
> Shouldn't it be `1` only when compiling in no release mode? Possibly this will prevent some optimization.

I just checked with different values when in release mode and LL file and binary are identical irrespective of the flag.